File: VB/DXBindingExample/ViewModel/TwoWayViewModel.vb
Imports DevExpress.Mvvm.DataAnnotations

Namespace DXBindingExample.ViewModel
    <POCOViewModel>
    Public Class TwoWayViewModel
        Inherits BaseViewModel

        Public Overrides ReadOnly Property Title() As Object
            Get
                Return "TwoWay Binding"
            End Get
        End Property
        Public Overridable Property FirstName() As String
        Public Overridable Property LastName() As String
        Public Sub New()
            FirstName = "John"
            LastName = "Smith"
        End Sub
        Public Function Sum(ByVal x As Integer, ByVal y As Integer) As Integer
            Return x + y
        End Function
    End Class
End Namespace

File: CS/DXBindingExample/ViewModel/TwoWayViewModel.cs
using DevExpress.Mvvm.DataAnnotations;

namespace DXBindingExample.ViewModel {
    [POCOViewModel]
    public class TwoWayViewModel : BaseViewModel {
        public override object Title { get { return "TwoWay Binding"; } }
        public virtual string FirstName { get; set; }
        public virtual string LastName { get; set; }
        public TwoWayViewModel() {
            FirstName = "John";
            LastName = "Smith";
        }
        public int Sum(int x, int y) {
            return x + y;
        }
    }
}
